client/src/planwise/client/coverage.cljs

```diff
@@ -81,15 +81,20 @@
 
 (defn coverage-algorithm-filter-options
   [{:keys [coverage-algorithm value on-change empty disabled?]}]
-  (let [list      (rf/subscribe [:coverage/supported-algorithms])
-        criteria  (get-in @list [(keyword coverage-algorithm) :criteria])]
+  (let [list         (rf/subscribe [:coverage/supported-algorithms])
+        criteria     (get-in @list [(keyword coverage-algorithm) :criteria])
+        valid-keys   (keys criteria)
+        update-value (fn [key change]
+                       (if (= coverage-algorithm "drive-walk-friction")
```

Comment:
I don't think you need this `if` here (plus the special case handling should is a smell). In fact, the previous behavior was wrong in that it only allowed a single criterion. Filtering the keys (using `select-keys`) is the correct and general solution.

Reply:
I'll fix it.